designer/src/com/android/tools/idea/common/property/editors/EnumEditor.java

```diff
@@ -466,17 +466,20 @@ private void setBorders() {
 
     @Override
     public void setUI(ComboBoxUI ui) {
-      myUseDarculaUI = !(ui instanceof WindowsComboBoxUI) && !ApplicationManager.getApplication().isUnitTestMode();
```

Comment:
Instead of adding `if (SystemUtils.IS_OS_WINDOWS)` please change this line to  
```
myUseDarculaUI = ~(ui instanceof BasicComboBoxUI) && !ApplicationManager.getApplication().isUnitTestMode();  
```  
That way the CustomDarculaComboBoxUI(this) logic can continue as normal.

Reply:
I'm pretty sure that they used `com.sun.java.swing.plaf.windows.WindowsComboBoxUI` instead of `javax.swing.plaf.basic.BasicComboBoxUI` because this is a ugly hack to hardcode `CustomDarculaComboBoxUI` only on Windows since `WindowsComboBoxUI`, or else they won't import `WindowsComboBoxUI` since it's an internal API that is undocumented and unsupported and it's very bad practice to use internal APIs. Since I don't know the reason why they decided to check for `WindowsComboBoxUI` instead of checking if the OS is Windows and then checking for `BasicComboBoxUI`, I'm just going to keep the check for `WindowsComboBoxUI`.